### DISCUSSION TITLE
Is there a step-by-step instruction or article on how to add your own package?

### DISCUSSION THREAD

**Discussion by farcop** [wolfi-dev/os#30606] [Q&A] [answered]
I want to add a package for wolfi-os, is there a step-by-step instruction on how to do this?

**Comment by kranurag7** [comment, marked as answer]
ref: https://edu.chainguard.dev/open-source/wolfi/building-a-wolfi-package/ and other examples in this repository. 
make targets is pretty useful when you're developing packages. 

Usually the workflow abstraction is that you write your melange config in foo.yaml and considering if your package name is foo then you do the following from root of the repository:

```bash
make dev-container
make package/foo # for building your apk
make test/foo # for testing
```
Hope this helps.

**Reply by farcop** [reply]
Thank you!

**Reply by farcop** [reply]
Added my first package https://github.com/wolfi-dev/os/issues/30607
